src/common/updaters/ThemeFromUrlUpdater/index.tsx

```diff
@@ -0,0 +1,34 @@
+import { useEffect, useRef } from 'react'
+
+import { useLocation } from 'react-router-dom'
+
+import { useAppDispatch } from 'legacy/state/hooks'
+import { useIsDarkMode } from 'legacy/state/user/hooks'
+import { updateUserDarkMode } from 'legacy/state/user/reducer'
+
+/**
+ * Switch theme depending on theme query parameter
+ */
+export function ThemeFromUrlUpdater() {
+  const darkMode = useIsDarkMode()
+  const darkModeRef = useRef(darkMode)
+  const dispatch = useAppDispatch()
+  const { search } = useLocation()
+
+  darkModeRef.current = darkMode
+
+  useEffect(() => {
+    const searchParams = new URLSearchParams(search)
+    const themeValue = searchParams.get('theme')
+
+    if (!themeValue) return
+
+    const isDarkMode = searchParams.get('theme') === 'dark'
+
+    if (isDarkMode === darkModeRef.current) return
+
+    dispatch(updateUserDarkMode({ userDarkMode: isDarkMode }))
```

Comment:
in some other params, we remove the param from the URL once used. Not saying we should do it here, just wondering if it makes sense

Reply:
For now this param is only for widget where we don't see a URL of iframe. But I see what you mean, probably we would need to clean up the URL